File: day2/part2.py
#!/usr/bin/env python

def diff2(a,b):
	common = []
	for i in range(len(a)):
		if a[i] == b[i]:
			common.append(a[i])
	return common

with open("input.txt", "r") as fh:
	input = fh.read()

lines = input.split("\n")
for i in range(len(lines)):
	for j in range(len(lines)):
		if i == j or lines[i] == '' or lines[j] == '':
			continue
		common = diff2(lines[i], lines[j])
		if (len(lines[i]) - len(common) == 1):
			print("--")
			print(lines[i])
			print(lines[j])
			print(''.join(common))
			exit(1)

File: day2/part2.swift
#!/usr/bin/swift

import Foundation

let textFile = try String(contentsOf: URL(fileURLWithPath: "input.txt")) 
let lines = textFile.split(separator: "\n")
let lineLength = lines[0].count
var commonWord:String?

for i in 0...lineLength {
	var s = Set<String>()
	for word in lines {
		var x = word
		let inx = word.index(word.startIndex, offsetBy:i)
		x.remove(at: inx)
		let subword = String(x)
		if s.contains(subword) {
			commonWord = subword
			break
		}
		s.insert(subword)
	}
	if commonWord != nil { break }
}

if commonWord != nil { 
	print("commonWord: \(commonWord!)")
}
